libcudacxx/include/cuda/std/detail/libcxx/include/span

```diff
@@ -223,23 +212,51 @@ _CCCL_CONCEPT_FRAGMENT(
     requires(_CUDA_VRANGES::sized_range<_Range>),
     requires((_CUDA_VRANGES::borrowed_range<_Range> || _CCCL_TRAIT(is_const, _ElementType))),
     requires((!_CCCL_TRAIT(is_array, remove_cvref_t<_Range>))),
-    requires((!__is_std_span<remove_cvref_t<_Range>> && !__is_std_array<remove_cvref_t<_Range>>
-              && !__is_std_initializer_list<remove_cvref_t<_Range>>) ),
+    requires((!__is_std_span<remove_cvref_t<_Range>> && !__is_std_array<remove_cvref_t<_Range>>) ),
     requires(_CCCL_TRAIT(
       is_convertible, remove_reference_t<_CUDA_VRANGES::range_reference_t<_Range>> (*)[], _ElementType (*)[]))));
 
 template <class _Range, class _ElementType>
 _CCCL_CONCEPT __span_compatible_range = _CCCL_FRAGMENT(__span_compatible_range_, _Range, _ElementType);
 
-#    if _CCCL_STD_VER >= 2020
+#  else // // ^^^ !_CCCL_COMPILER(MSVC2017) ^^^ / vvv _CCCL_COMPILER(MSVC2017) vvv
+
+template <class _Range, class _ElementType, class = void>
+_CCCL_INLINE_VAR constexpr bool __span_compatible_range = false;
+
+template <class _Range, class _ElementType>
+_CCCL_INLINE_VAR constexpr bool __span_compatible_range<
+  _Range,
+  _ElementType,
+  void_t<
+    // // is a contiguous range
+    // enable_if_t<_CUDA_VRANGES::contiguous_range<_Range>, nullptr_t>,
+    // // is a sized range
+    // enable_if_t<_CUDA_VRANGES::sized_range<_Range>, nullptr_t>,
+    // // is a borrowed range or ElementType is const
+    // enable_if_t<(_CUDA_VRANGES::borrowed_range<_Range> || _CCCL_TRAIT(is_const, _ElementType)), nullptr_t>,
+    // is not a C-style array
```

Comment:
Should we at least check for `decltype(_CUDA_VSTD::size(_CUDA_VSTD::declval<_Container&>())),`

To fake sized range